ext/mbstring/tests/gh10192_utf7.phpt

```diff
@@ -0,0 +1,542 @@
+--TEST--
+GH-10192 (mb_detect_encoding() results for UTF-7 differ between PHP 8.0 and 8.1)
+--EXTENSIONS--
+mbstring
+--FILE--
+<?php
+
+$testcases = [
+    'non-base64 character after +' => 'A + B',
+    'non-base64 character after -' => 'A - B',
+    'base64 character before +' => 'A 1+ B',
+    'base64 character before -' => 'A 1- B',
+    'base64 character after +' => 'A +1 B',
+    'base64 character after -' => 'A -1 B',
+    'base64 character before and after +' => 'A 1+1 B',
+    'base64 character before and after -' => 'A 1-1 B',
+    'string ends with +' => 'A +',
+    'string ends with -' => 'A -',
+    '+ and -' => 'A +- B',
+    '- and +' => 'A -+ B',
+    'valid direct encoding character =' => 'A = B',
+    'invalid direct encoding character ~' => 'A ~ B',
+    'invalid direct encoding character \\' => 'A \\ B',
+    'invalid direct encoding character ESC' => "A \x1b B",
+    'valid direct encoding character = after +' => 'A += B',
+    'invalid direct encoding character ~ after +' => 'A +~ B',
+    'invalid direct encoding character \\ after +' => 'A +\\ B',
+    'invalid direct encoding character ESC after +' => "A +\x1b B",
+    'valid base64 character between + and -' => 'A +ZeVnLIqe- B', // 日本語 in UTF-16BE
+    'invalid base64 character between + and -' => 'A +ZeVnLIq- B', // 日本語 in UTF-16BE without the last character
+    'valid base64 character between + and non-base64 character' => 'A +ZeVnLIqe B',
+    'invalid base64 character between + and non-base64 character' => 'A +ZeVnLIq B',
+    'valid base64 character between + and base64 character' => 'A +ZeVnLIqe1 B',
+    'invalid base64 character between + and base64 character' => 'A +ZeVnLIq1 B',
+    'valid base64 character between + and end of string' => 'A +ZeVnLIqe',
+    'invalid base64 character between + and end of string' => 'A +ZeVnLIq',
+    'valid base64 character consisting only of + between + and -' => 'A +++++++++- B',
+    'invalid base64 character consisting only of + between + and -' => 'A +++++++++- B',
+    'valid base64 character consisting only of + between + and non-base64 character' => 'A +++++++++ B',
+    'invalid base64 character consisting only of + between + and non-base64 character' => 'A +++++++++ B',
+    'valid base64 character consisting only of + between + and base64 character' => 'A +++++++++1 B',
+    'invalid base64 character consisting only of + between + and base64 character' => 'A +++++++++1 B',
+    'valid base64 character consisting only of + between + and end of string' => 'A +++++++++',
+    'invalid base64 character consisting only of + between + and end of string' => 'A +++++++++',
+    'valid base64 character using surrogate pair between + and -' => 'A +2GfePQ- B', // 𩸽 in UTF-16BE
+    'first 16 bits of base64 character using surrogate pair between + and -' => 'A +2Gc- B', // first 16 bits of 𩸽 in UTF-16BE
```

Comment:
@alexdowad 
I think the test case is here.
Both `mb_check_encoding` and `mb_convert_encoding` recognize it as an invalid string 😊